### PR TITLE
Cloud Workstations - Workstation Cluster

### PR DESCRIPTION
**Pull request by bschaatsbergen**
Closes [#12763](https://github.com/hashicorp/terraform-provider-google/issues/12763)

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
google_workstations_workstation_cluster
```

### CODE REVIEW

File: mmv1/products/workstations/api.yaml
Comment (DrFaust92) [inline mmv1/products/workstations/api.yaml:107]: output: true

**Comment by bschaatsbergen** [comment]
Hi @ScottSuarez, I believe that most of it is done for this PR. 
The other 2 resources for Cloud Workstations are offloaded to other PRs (we can address those later, but should preferably go together in the same release).

Could you give me a hand with the failing acceptance tests? I'm unable to run `make testacc` for some reason.

File: mmv1/templates/terraform/examples/workstation_cluster_private.tf.erb
Comment (ScottSuarez) [inline mmv1/templates/terraform/examples/workstation_cluster_private.tf.erb:28]: You'll need to set `auto_create_subnetwork` to false here. Otherwise there is collision with the resource below.

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_network#auto_create_subnetworks

File: mmv1/products/workstations/api.yaml
Comment (ScottSuarez) [inline mmv1/products/workstations/api.yaml:34]: Could you add a test that flexes update functionality? It would be a handwritten test.

File: mmv1/products/workstations/api.yaml
Comment (ScottSuarez) [inline mmv1/products/workstations/api.yaml:105]: could you flex this field in your tests?

File: mmv1/products/workstations/api.yaml
Comment (ScottSuarez) [inline mmv1/products/workstations/api.yaml:123]: This should be of type fingerprint

File: mmv1/products/workstations/api.yaml
Comment (ScottSuarez) [inline mmv1/products/workstations/api.yaml:142]: These two fields are irrelevant and can be removed.

File: mmv1/products/workstations/api.yaml
Comment (ScottSuarez) [inline mmv1/products/workstations/api.yaml:185]: output:true

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

File: mmv1/products/workstations/api.yaml
Comment (bschaatsbergen) [reply mmv1/products/workstations/api.yaml:185]: Fixed

File: mmv1/products/workstations/api.yaml
Comment (bschaatsbergen) [reply mmv1/products/workstations/api.yaml:142]: Are you referring to updateTime and deleteTime? If so, I have removed these :+1:

File: mmv1/products/workstations/api.yaml
Comment (bschaatsbergen) [reply mmv1/products/workstations/api.yaml:123]: Should `etag` be of type fingerprint? Which field are you exactly referring to?

File: mmv1/templates/terraform/examples/workstation_cluster_private.tf.erb
Comment (bschaatsbergen) [reply mmv1/templates/terraform/examples/workstation_cluster_private.tf.erb:28]: Fixed

File: mmv1/products/workstations/api.yaml
Comment (bschaatsbergen) [reply mmv1/products/workstations/api.yaml]: Done

File: mmv1/products/workstations/api.yaml
Comment (bschaatsbergen) [reply mmv1/products/workstations/api.yaml:34]: Added a test, see `resource_workstations_workstation_cluster_test.go.erb`.

File: mmv1/products/workstations/api.yaml
Comment (bschaatsbergen) [reply mmv1/products/workstations/api.yaml:105]: I've set `display_name` in the `TestAccorkstationsWorkstationCluster_update` test.

**Comment by bschaatsbergen** [comment]
@ScottSuarez, the above comments are addressed, could you please check for me why the tests are failing?

File: mmv1/products/workstations/api.yaml
Comment (ScottSuarez) [reply mmv1/products/workstations/api.yaml]: correct ! etag!

File: mmv1/products/workstations/api.yaml
Comment (ScottSuarez) [reply mmv1/products/workstations/api.yaml]: correct !

**Comment by ScottSuarez** [comment]
Looks like I misspelt the field so the test failed.
```hcl
=== RUN   TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample
=== PAUSE TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample
=== CONT  TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample
    provider_test.go:315: Step 1/2 error: Error running pre-apply refresh: exit status 1
        
        Error: Unsupported argument
        
          on terraform_plugin_test.tf line 26, in resource "google_compute_network" "default":
          26:   auto_create_subnetwork = false
        
        An argument named "auto_create_subnetwork" is not expected here. Did you mean
        "auto_create_subnetworks"?
--- FAIL: TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample (0.46s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-google-beta/google-beta	0.494s
FAIL
```

If you could try to run these tests locally that would help. What obstacles do you see in way of running them locally?

**Comment by bschaatsbergen** [comment]
@ScottSuarez, for some reason when I try to run the tests I get a bunch of errors in unrelated files, e.g. BigQuery, Cloud Run V2 etc. I'll see if I can figure out, if you can meanwhile share the error logs of the failed test I can try to figure out what's wrong.

**Comment by bschaatsbergen** [comment]
Hmm, can you share the build log with me please @ScottSuarez 

> The provider crashed while running the VCR tests in REPLAYING mode Please fix it to complete your PR View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7005/artifacts/107c62a0-ac79-421a-8d82-97c4721cc582/build-log/replaying_test.log)

**Comment by ScottSuarez** [comment]
Could you pull in the latest changes from main? This is a known issue that has been fixed.

**Comment by bschaatsbergen** [comment]
> Could you pull in the latest changes from main? This is a known issue that has been fixed.

On it!

**Comment by bschaatsbergen** [comment]
Hmm @ScottSuarez :point_up:, what error is being reported?

**Comment by ScottSuarez** [comment]
```
        Error: Error creating WorkstationCluster: googleapi: Error 400: Malformed name: 'projects/ci-test-project-188019/locations/us-central1/workstationClusters/'
        Details:
        [
          {
            "@type": "type.googleapis.com/google.rpc.BadRequest",
            "fieldViolations": [
              {
                "description": "URL path has empty component",
                "field": "projects/ci-test-project-188019/locations/us-central1/workstationClusters/"
              }
            ]
          }
        ]
        
          with google_workstations_workstation_cluster.default,
          on terraform_plugin_test.tf line 3, in resource "google_workstations_wor
     ```

**Comment by bschaatsbergen** [comment]
I guess it's due to the casing typo, fixing now - lets see.

**Comment by bschaatsbergen** [comment]
@ScottSuarez, can you take a look at my latest commit and push me into the right direction? I'm unable to run the test locally for some reason.. I'll see if I can figure my local testing out tomorrow.

**Comment by bschaatsbergen** [comment]
I'll go and figure out tomorrow morning why I still can't run the tests locally to prevent this back and forth 😄. Will need it anyways for the upcoming Cloud Workstation resources.

**Comment by bschaatsbergen** [comment]
@ScottSuarez, for some reason when I generate the beta provider the makefile seems to be missing which blocks me from running any acceptance tests..

**Comment by bschaatsbergen** [comment]
@ScottSuarez, tests are running locally fine. They just fail in the post-destroy:

```console
make testacc TEST=./google-beta TESTARGS='-run=TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample'
==> Checking that code complies with gofmt requirements...
go vet
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample
=== PAUSE TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample
=== CONT  TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample
    testing_new.go:84: Error running post-test destroy, there may be dangling resources: exit status 1

        Error: Error when reading or editing Subnetwork: googleapi: Error 400: The subnetwork resource 'projects/playground-bruno/regions/us-east1/subnetworks/tf-test-workstation-clusterbip9znxbgx' is already being used by 'projects/playground-bruno/regions/us-east1/addresses/workstations-cluster-v7ie2mwou5ftasynrtwavsfu4q', resourceInUseByAnotherResource

--- FAIL: TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample (3024.22s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-google-beta/google-beta	3024.830s
FAIL
make: *** [testacc] Error 1
```

**Comment by bschaatsbergen** [comment]
All tests pass locally @ScottSuarez can you please check what's reported in the VCR-test check?

File: mmv1/products/workstations/api.yaml
Comment (ScottSuarez) [inline mmv1/products/workstations/api.yaml:119]: This field can likely be omitted.. Not sure how useful it will be to terraform users. Let me know if you disagree

**Review by ScottSuarez — CHANGES_REQUESTED** [review]
The test failure was due to a quota issue... only 5 of these can be provisioned at a time so dangling resources from the undeleted previous test runs were blocking... 

I've cleared these resources. /gcbrun

File: mmv1/products/workstations/api.yaml
Comment (bschaatsbergen) [reply mmv1/products/workstations/api.yaml:119]: I'll remove this

File: mmv1/third_party/terraform/tests/resource_workstations_workstation_cluster_test.go.erb
Comment (bschaatsbergen) [reply mmv1/third_party/terraform/tests/resource_workstations_workstation_cluster_test.go.erb:55]: Will do 👍🏼

Comment (bschaatsbergen) [reply mmv1/third_party/terraform/tests/resource_workstations_workstation_cluster_test.go.erb:55]: Fixed.

**Comment by bschaatsbergen** [comment]
@ScottSuarez, all tests work locally.

```
 ~/go/src/github.com/hashicorp/terraform-provider-google-beta   main ±  make testacc TEST=./google-beta TESTARGS='-run=TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample'
==> Checking that code complies with gofmt requirements...
go vet
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample
=== PAUSE TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample
=== CONT  TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample
--- PASS: TestAccWorkstationsWorkstationCluster_workstationClusterBasicExample (3074.95s)
PASS
ok  	github.com/hashicorp/terraform-provider-google-beta/google-beta	3075.576s
 ~/go/src/github.com/hashicorp/terraform-provider-google-beta   main ±  make testacc TEST=./google-beta TESTARGS='-run=TestAccWorkstationsWorkstationCluster_workstationClusterPrivateExample'
==> Checking that code complies with gofmt requirements...
go vet
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccWorkstationsWorkstationCluster_workstationClusterPrivateExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccWorkstationsWorkstationCluster_workstationClusterPrivateExample
=== PAUSE TestAccWorkstationsWorkstationCluster_workstationClusterPrivateExample
=== CONT  TestAccWorkstationsWorkstationCluster_workstationClusterPrivateExample
--- PASS: TestAccWorkstationsWorkstationCluster_workstationClusterPrivateExample (3052.35s)
PASS
ok  	github.com/hashicorp/terraform-provider-google-beta/google-beta	3052.945s
 ~/go/src/github.com/hashicorp/terraform-provider-google-beta   main ±  make testacc TEST=./google-beta TESTARGS='-run=TestAccorkstationsWorkstationCluster_update'
==> Checking that code complies with gofmt requirements...
go vet
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccorkstationsWorkstationCluster_update -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccorkstationsWorkstationCluster_update
=== PAUSE TestAccorkstationsWorkstationCluster_update
=== CONT  TestAccorkstationsWorkstationCluster_update
--- PASS: TestAccorkstationsWorkstationCluster_update (9272.82s)
PASS
ok  	github.com/hashicorp/terraform-provider-google-beta/google-beta	9273.451s
```

**Comment by ScottSuarez** [comment]
The delete operation is timing out... Is there a way to add an optional field for a soft delete (ie: delete but ignore pending status).

Unsure about the environment differences and why our delete operations are taking drastically longer.

**Comment by bschaatsbergen** [comment]
> The delete operation is timing out... Is there a way to add an optional field for a soft delete (ie: delete but ignore pending status).
> 
> 
> 
> Unsure about the environment differences and why our delete operations are taking drastically longer. 

Hmm that's crap, considering that this resource will be the parent of other resources (e.g. a workstation, and a cluster configuration) we might be able to pull this off. Do you have any reference implementations that use soft deletes?

**Comment by ScottSuarez** [comment]
/gcbrun

Let me try to rerun before we go about this. Possible its a transient issue... I also would like to contact the team if it is an issue before proceeding down this path.

**Comment by bschaatsbergen** [comment]
> /gcbrun
> 
> 
> 
> Let me try to rerun before we go about this. Possible its a transient issue... I also would like to contact the team if it is an issue before proceeding down this path. 

Right, thanks for checking it 👍🏼

**Comment by bschaatsbergen** [comment]
Neat, @ScottSuarez looks like this worked - besides that (likely) the update test now timed out (?)

**Comment by ScottSuarez** [comment]
```

=== CONT  TestAccorkstationsWorkstationCluster_update
    provider_test.go:315: Step 1/5 error: After applying this test step and performing a `terraform refresh`, the plan was not empty.
        stdout
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          + create
        -/+ destroy and then create replacement
        
        Terraform will perform the following actions:
        
          # google_compute_subnetwork.default will be created
          + resource "google_compute_subnetwork" "default" {
              + creation_timestamp         = (known after apply)
              + external_ipv6_prefix       = (known after apply)
              + fingerprint                = (known after apply)
              + gateway_address            = (known after apply)
              + id                         = (known after apply)
              + ip_cidr_range              = "10.0.0.0/24"
              + ipv6_cidr_range            = (known after apply)
              + name                       = "tf-test-workstation-clusterkowd4p0wlv"
              + network                    = "tf-test-workstation-clusterkowd4p0wlv"
              + private_ip_google_access   = (known after apply)
              + private_ipv6_google_access = (known after apply)
              + project                    = (known after apply)
              + purpose                    = (known after apply)
              + region                     = "us-central1"
              + secondary_ip_range         = (known after apply)
              + self_link                  = (known after apply)
              + stack_type                 = (known after apply)
            }
        
          # google_workstations_workstation_cluster.default must be replaced
        -/+ resource "google_workstations_workstation_cluster" "default" {
              ~ conditions             = [
                  - {
                      - code    = 5
                      - details = []
                      - message = "reconcile endpoint: create: googleapi: Error 404: The resource 'projects/xxxxx/regions/us-central1/subnetworks/tf-test-workstation-clusterkowd4p0wlv' was not found, notFound"
                    },
                ] -> (known after apply)
              ~ create_time            = "2023-01-17T21:13:48.595856163Z" -> (known after apply)
              ~ degraded               = true -> (known after apply)
              ~ etag                   = "ppY12CyH_l0nqhdqArmVtt135q5kqhXPumcwT0Dk-ms" -> (known after apply)
              ~ id                     = "projects/xxxxxx/locations/us-central1/workstationClusters/tf-test-workstation-clusterkowd4p0wlv" -> (known after apply)
              ~ name                   = "projects/xxxxxx/locations/us-central1/workstationClusters/tf-test-workstation-clusterkowd4p0wlv" -> (known after apply)
              ~ project                = "xxxxxx" -> (known after apply)
              - reconciling            = false -> null
              ~ subnetwork             = "projects/xxxxxx/regions/us-central1/subnetworks/tf-test-workstation-clusterkowd4p0wlv" -> (known after apply) # forces replacement
              ~ uid                    = "eb16edb0-d400-4e1e-ae3d-4bb30d9fa911" -> (known after apply)
                # (5 unchanged attributes hidden)
            }
        
        Plan: 2 to add, 0 to change, 1 to destroy.
--- FAIL: TestAccorkstationsWorkstationCluster_update (3009.73s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-google-beta/google-beta	3009.768s
FAIL
```

**Comment by bschaatsbergen** [comment]
hmm I'm not quite sure how this might happen @ScottSuarez ?

**Comment by ScottSuarez** [comment]
/gcbrun

I have a thought about why this may be happening.. Lets try a rerun !

**Comment by mar-pan** [comment]
Hello, is there any chance to finish this PR anytime soon? I would love to use this resources in my project

**Comment by bschaatsbergen** [comment]
@ScottSuarez any idea regarding the failing update test? It seems to run fine locally, takes a while though.

**Comment by ScottSuarez** [comment]
/gcbrun 

we can try to rerun this.. Looks like the failure is due to a timeout. The test is taking >90minutes to complete
```
=== CONT  TestAccorkstationsWorkstationCluster_update
panic: test timed out after 1h30m0s

````

**Comment by bschaatsbergen** [comment]
Creating or Deleting a cluster takes roughly 30 minutes. I believe that the update does a create and delete twice. 

We could optionally bump the timeout to 180 min to be on the safe side for this test, or we simply remove the test.

**Comment by ScottSuarez** [comment]
Ah gotcha! I would say lets change the update operation to only delete once... We can use the update test to test update functionality, not forcenew behavior

**Comment by ScottSuarez** [comment]
> Creating or Deleting a cluster takes roughly 30 minutes. I believe that the update does a create and delete twice.
> 
> We could optionally bump the timeout to 180 min to be on the safe side for this test, or we simply remove the test.

hey just checking in.. no rush or anything. I'm available if there is anything I can help with.

**Comment by bschaatsbergen** [comment]
Checking this today @ScottSuarez, will ping you if I bunp into anything

**Comment by bschaatsbergen** [comment]
In terms of rolling out Cloud Workstations, there's 2 other resources that need to be created. I had prepped PRs for them, but they are waiting for a rebase once this is merged.

- https://github.com/GoogleCloudPlatform/magic-modules/pull/7017
- https://github.com/GoogleCloudPlatform/magic-modules/pull/7018

I can add these resources also to this PR, otherwise I suppose we need to prioritise the above 2 PRs once this PR is merged.

What do you think @ScottSuarez ?

**Comment by ScottSuarez** [comment]
@bschaatsbergen, let's keep them separate since its easier to review resources in isolation then not. I'm going to a last pass on this to make sure everything is in order now.

**Review by ScottSuarez — CHANGES_REQUESTED** [review]
Last small set of changes/suggestions

**Comment by ScottSuarez** [comment]
Ah I just noticed we removed the update test?

Apologies for the confusion.. I was suggesting to only update fields in the test. Not tear down and recreate. We'll need that test to be readded. 

So the update test should be as follows

Deploy x configurations..
`update` the maximum set of fields that /can/ be updated. If a field has forcenew it will require teardown. Given the long deployment time lets not update these fields.

**Comment by bschaatsbergen** [comment]
I'll jump on the comments tomorrow @ScottSuarez, though I tried making such a test but it still forced recreation, even if I simply added a label.

**Comment by ScottSuarez** [comment]
> I'll jump on the comments tomorrow @ScottSuarez, though I tried making such a test but it still forced recreation, even if I simply added a label. Do you have any suggestions?

That doesn't sound right !! 

I would try to use the provider locally and go through the deployment then do a terraform plan to see why recreation is being first.

You could also turn on logging for the test on your local machine with `TF_LOG=debug` and `TF_LOG_PATH=<somepath>/temp.log`
https://developer.hashicorp.com/terraform/internals/debugging

**Comment by bschaatsbergen** [comment]
@ScottSuarez, fixed the comments you put down. My assumption that the update in place wasn't working was incorrect. I just the provider locally and the update only takes a few seconds.

Can it be that because we use 2 different test configs (also 2 random suffixes or not?), it does recreate as part of the update test and therefor exceeds the timeout?

Feel free to commit something in my branch btw.

File: mmv1/third_party/terraform/tests/resource_workstations_workstation_cluster_test.go.erb
Comment (ScottSuarez) [inline mmv1/third_party/terraform/tests/resource_workstations_workstation_cluster_test.go.erb:75]: The name of the subnetwork isn't matching up with generated name of the subnetwork in `testAccWorkstationsWorkstationCluster_workstationClusterBasicExample`... This is forcing recreation here causing the `google_workstations_workstation_cluster` to subsequently get recreated. Lets either align the names or encapsulate both configs (before and after) within this file explicitly. 

[modular-magician/terraform-provider-google-beta@`auto-pr-7005-old..auto-pr-7005`#diff-38fbf90791](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7005-old..auto-pr-7005#diff-38fbf907918e59c24a64079871d90f13c5f1d95b93924b38b483dd2dc48fc902R81)

**Comment by ScottSuarez** [comment]
I pushed changes that should align the configurations

File: mmv1/third_party/terraform/tests/resource_workstations_workstation_cluster_test.go.erb
Comment (bschaatsbergen) [reply mmv1/third_party/terraform/tests/resource_workstations_workstation_cluster_test.go.erb]: Sharp.. thanks Scott

**Comment by bschaatsbergen** [comment]
It still failed for some reason, thanks for the fix @ScottSuarez

**Review by ScottSuarez — APPROVED** [review]
lgtm thanks for your patience and persistence!

**Comment by bschaatsbergen** [comment]
Sweet! @ScottSuarez I'm going to immediately hop on the other Cloud Workstation resources.

**Comment by bschaatsbergen** [comment]
I noticed a mistake here:

image

`[cluster_hostname](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/workstations_workstation_cluster#cluster_hostname)` and `[service_attachment_uri](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/workstations_workstation_cluster#service_attachment_uri)` are output only fields.

What do you recommend we do? Remove these in a major, or remove them now (might be a breaking change if we remove them now?)

@rileykarson / @SarahFrench

**Comment by SarahFrench** [comment]
I don't think there's an issue with the provider, it's just that the documentation is a little misleading here. [You marked those fields as outputs correctly](https://github.com/bschaatsbergen/magic-modules/blob/35abf1be681cecb8bc3bdbbc3ed6ac286f6b0466/mmv1/products/workstations/api.yaml#L138-L151) and the [resulting schema has those fields marked as Computed](https://github.com/hashicorp/terraform-provider-google-beta/blob/7141d88774fdcec4482b2ccf46a855dfc8814fd0/google-beta/resource_workstations_workstation_cluster.go#L101-L114) as we'd expect.

The documentation misleading here because the `parent_cluster_config` block contains outputs, but it also includes a field that can be set in the user's config (`enable_private_endpoint`). This means the `parent_cluster_config` block is included in the Arguments section because it contains something that a user can set, and `cluster_hostname` and `service_attachment_uri` aren't displayed under Attributes as a result. However, yeah, overall it makes the other fields look like they can be set by users.

We could address this by changing the template that generates these docs, e.g. to the idea below, but it'll have wide impacts and may need to be discussed by the maintainers about the correct approach:

```
The private_cluster_config block supports:

enable_private_endpoint - (Required) Whether Workstations endpoint is private.

cluster_hostname - (Output Only) Hostname for the workstation cluster. This field will be populated only when private endpoint is enabled. To access workstations in the cluster, create a new DNS zone mapping this domain name to an internal IP address and a forwarding rule mapping that address to the service attachment.

service_attachment_uri - (Output Only) Service attachment URI for the workstation cluster. The service attachemnt is created when private endpoint is enabled. To access workstations in the cluster, configure access to the managed service using (Private Service Connect)[https://cloud.google.com/vpc/docs/configure-private-service-connect-services].

```

👆 After saying all that, the tl;dr is I think it's our problem instead of yours! 😁

**Comment by bschaatsbergen** [comment]
Thanks for elaborating @SarahFrench, you're right; it seems like the docs are misleading - I would think that they are arguments to pass. Anyhow, something we can fix in generating the docs correctly 👍🏼

**Pull request merged**
